File: python_problems/array_strings/increase_triplet_subsequence.py
"""
Given an integer array nums, return true if there exists a triple of indices (i, j, k) such that i < j < k and nums[i] < nums[j] < nums[k]. If no such indices exists, return false.

 

Example 1:

Input: nums = [1,2,3,4,5]
Output: true
Explanation: Any triplet where i < j < k is valid.
Example 2:

Input: nums = [5,4,3,2,1]
Output: false
Explanation: No triplet exists.
Example 3:

Input: nums = [2,1,5,0,4,6]
Output: true
Explanation: The triplet (3, 4, 5) is valid because nums[3] == 0 < nums[4] == 4 < nums[5] == 6.
"""
from typing import List

class Solution:
    def increasingTriplet(self, nums):
        min1 = float('inf')
        min2 = float('inf')
        for n in nums:
            if n <= min1:
                min1 = n  # Update first minimum
            elif n <= min2:
                min2 = n  # Update second minimum
            else:
                return True  # Found a third number greater than both
        return False  # No triplet found

File: Go/array_strings/increase_triplet_subsequence.go
package array_strings

func IncreasingTriplet(nums []int) bool {
	min1, min2 := int(^uint(0)>>1), int(^uint(0)>>1) // Inicializar a infinito (max int)
	for _, n := range nums {
		if n <= min1 {
			min1 = n
		} else if n <= min2 {
			min2 = n
		} else {
			return true
		}
	}
	return false
}
